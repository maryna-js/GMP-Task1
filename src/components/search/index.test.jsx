import React from "react";
import SearchPanel from "./index";
import { render, fireEvent, screen } from "@testing-library/react";

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockHistoryPush,
    }),
}));


describe("SearchPanel", () => {
    const mockSetSearchValue = jest.fn();


    const renderSearchPanel = () =>
        render(
            <SearchPanel
                setSearchValue={mockSetSearchValue}
            />
        );

    it("can render", () => {
        const { asFragment } = renderSearchPanel();
        expect(asFragment()).toMatchSnapshot();
    });

    it("can set value by input change", () => {
        renderSearchPanel();
        fireEvent.change(screen.getByPlaceholderText("What do you want to watch?"), {
            target: { value: "a" },
        });

        expect(mockSetSearchValue);
    });

    it("can set value by input change", () => {
        renderSearchPanel();
        fireEvent.click(screen.getByText("Search"));
        expect(mockSetSearchValue);
    });
});

