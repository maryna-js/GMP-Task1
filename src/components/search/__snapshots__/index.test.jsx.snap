// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel can render 1`] = `
<DocumentFragment>
  <div
    class="wrapper-container"
  >
    <h3
      class="heading"
    >
      Find your movie
    </h3>
    <div
      class="search-wrapper"
    >
      <input
        class="search-input"
        placeholder="What do you want to watch?"
      />
      <button
        class="search-button"
      >
        Search
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SearchPanel can set value by input change 1`] = `[MockFunction]`;

exports[`SearchPanel should render properly 1`] = `
<DocumentFragment>
  <div
    class="wrapper-container"
  >
    <h3
      class="heading"
    >
      Find your movie
    </h3>
    <div
      class="search-wrapper"
    >
      <input
        class="search-input"
        placeholder="What do you want to watch?"
      />
      <button
        class="search-button"
      >
        Search
      </button>
    </div>
  </div>
</DocumentFragment>
`;
